[WIP] orbs docs rework (#4506)
circleci/circleci-docs: jekyll/_cci2/orbs-faq.md

This document describes various questions and technical issues that you may find helpful when working with orbs.

## Private orbs

* **Question:** Can orbs be made private?

* **Answer:** Currently, all orbs published to the [Orb Registry]() are open source. Please subscribe to this feature request for updates: [Feature Request: private orbs](https://ideas.circleci.com/ideas/CCI-I-606)

## Difference between commands and jobs

* **Question:** What is the difference between commands and jobs?

> * **Answer:** Both [commands]({{site.baseurl}}/2.0/reusing-config/#the-commands-key) and [jobs]({{site.baseurl}}/2.0/reusing-config/#authoring-parameterized-jobs) are elements that can be used within orbs. _Commands_ contain one or many [steps]({{site.baseurl}}/2.0/configuration-reference/#steps), which contain the logic of the orb. Commands generally execute some shell code (bash). _Jobs_ are a definition of what steps/commands to run and within what [executor]({{site.baseurl}}/2.0/reusing-config/#executors) to run them. _Commands_ are invoked at the job level. _Jobs_ are invoked at the _[Workflow]({{site.baseurl}}/2.0/workflows/#workflows-configuration-examples)_ level.

Reviewer: _Jobs_ are a definition of what steps/commands to run and within -a predefined [executor]- ?

"Commands invoked at the job level" and "Jobs are invoked at the workflow level" confuses me a bit. I guess I dont know what 'invoke' means here. 
Terrible suggestions: Orb commands are used in a project configs's job? Jobs are used in a project config's workflow? Commands are drop in replacements for steps in a job?
Also a link to jobs here: _Commands_ are invoked at the job level would be nice as well.

Author: Thanks for this feedback! I will think on how best to clear this up...

Author: I've tried to improve this now

* **Answer:** Both [commands]({{site.baseurl}}/2.0/reusing-config/#the-commands-key) and [jobs]({{site.baseurl}}/2.0/reusing-config/#authoring-parameterized-jobs) are elements that can be used within orbs. _Commands_ contain one or many [steps]({{site.baseurl}}/2.0/configuration-reference/#steps), which contain the logic of the orb. Commands generally execute some shell code (bash). _Jobs_ are a definition of what steps/commands to run and within what [executor]({{site.baseurl}}/2.0/reusing-config/#executors) to run them. _Commands_ are invoked at the job level. _Jobs_ are invoked at the _[Workflow]({{site.baseurl}}/2.0/workflows/#workflows-configuration-examples)_ level.

```
workflows:
  info:
    jobs:
      - hello/hello-build
```
## Using Orbs on CircleCI Server

CircleCI publishes a web-based registry viewer so orbs documentation can be auto-generated. You can always pull the source of orbs directly as well. For example, you can run `circleci orb source circleci/hello-build@0.01`
* **Question:** Can orbs be used on a private installation of CircleCI server?

> ## Build Error When Testing Locally

Author: @KyleTryon just wondering about the reasons for removing most of the existing FAQs? If they are not obsolete in some way I would like to add them back in, in addition to the ones you've added here, as we've had good feedback on this doc in the past, suggesting some people have found them useful.

Reviewer 2: For this one here, we do not want to encourage testing locally in this way at the time. We are producing new testing documentation what will not make mention of running jobs locally. We will instead focus on testing code locally (which was previously not possible) and testing the orb itself directly on CircleCI (rather than a local circleci job). This also makes it easier for the CPE team to support this style of testing as we have less control over the local circleci build command.

Author: Ah yep that makes sense!

Author: My comment RE FAQs was regarding the entire doc, so the rest of the deleted FAQs too.

Author: @KyleTryon ^^

> * **Answer:** CircleCI Server does not yet support orbs natively. However, if you process your config prior to committing, orbs can be translated and used. We have put together a guide on using git pre-commit hooks to [use orbs on server](https://discuss.circleci.com/t/orbs-on-server-solution/36264)

Author: ```suggestion
* **Answer:** CircleCI Server does not yet support orbs natively. However, if you process your config prior to committing, orbs can be translated and used. Follow this guide on using git pre-commit hooks to [use orbs on server](https://discuss.circleci.com/t/orbs-on-server-solution/36264).
```
